main/test/flix/Test.Exp.Upcast.flix

```diff
@@ -128,4 +128,30 @@ namespace Test/Exp/Upcast {
                 g;
         ()
 
+    pub eff E
+
+    @test
+    def testUpcast13(): Unit =
+        let f = () -> 1;
+        let g = () -> 1 as \ E;
+        let _ =
+            if (true)
+                upcast f
+            else
+                g;
+        ()
+
+    @test
+    def testUpcast14(): Unit =
+        import new java.lang.StringBuilder(): ##java.lang.StringBuilder & Impure as newStringBuilder;
+        import new java.lang.Object(): ##java.lang.Object & Impure \ E as newObject;
+        let f = (_: ##java.lang.Object) -> newStringBuilder();
+        let g = (_: ##java.lang.StringBuilder) -> newObject();
+        let _ =
+            if (true)
+                upcast f
+            else
+                g;
+        ()
+
 }
```

Comment:
Can we add some more complex set formulas?

By the way, I think an easy test structure would be something like this:

```
(upcast (???: SomeType)): TargetType
```

Reply:
That doesn't work at the moment as we can only upcast effecst on functions. I'll get around later to upcasting expressions (in a new pr) :)